src/mondrian_forest/mondrian_tree_reg.rs

```diff
@@ -0,0 +1,684 @@
+use crate::common::{Classifier, ClfTarget, RegTarget, Regressor};
+use crate::mondrian_forest::alias::FType;
+use crate::mondrian_forest::mondrian_node::{NodeClassifier, NodeRegressor};
+use ndarray::{Array1, Array2};
+use num::{Float, FromPrimitive};
+use rand::prelude::*;
+use rand_distr::{Distribution, Exp};
+use std::collections::HashSet;
+use std::fmt;
+use std::usize;
+
+#[derive(Clone)]
+pub struct MondrianTreeRegressor<F: FType> {
+    n_features: usize,
+    rng: ThreadRng,
+    nodes: Vec<NodeRegressor<F>>,
+    root: Option<usize>,
+}
+
+impl<F: FType + fmt::Display> fmt::Display for MondrianTreeRegressor<F> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "")?;
+        if let Some(root) = self.root {
+            self.recursive_repr(Some(root), f, "", true)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+impl<F: FType + fmt::Display> MondrianTreeRegressor<F> {
+    fn recursive_repr(
+        &self,
+        node_idx: Option<usize>,
+        f: &mut fmt::Formatter<'_>,
+        prefix: &str,
+        is_last: bool,
+    ) -> fmt::Result {
+        if let Some(idx) = node_idx {
+            let node = &self.nodes[idx];
+            let node_prefix = if is_last { "└─" } else { "├─" };
+            let child_prefix = if is_last { "  " } else { "│ " };
+            let feature = if self.nodes[idx].feature > 100 {
+                String::from("_")
+            } else {
+                self.nodes[idx].feature.to_string()
+            };
+            writeln!(
+                f,
+                "{}{}Node {}: time={:.3}, min={:?}, max={:?}, thrs={:.2}, f={}, sums={}, counts={}",
```

Comment:
Just thinking out loud: it would be nice to leverage the same implementation you made for the classifier, although I know the written stuff is not the same